sdk/core/System.ClientModel/tests/client/ModelReaderWriter/Models/DiscriminatorSet/ModelX.cs

```diff
@@ -37,7 +38,7 @@ internal ModelX(string kind, string name, int xProperty, int? nullProperty, ILis
         {
             if (modelX == null)
             {
-                return null;
+                throw new ArgumentNullException(nameof(modelX));
```

Comment:
Non-nullable models throw ArgumentNullException in methods that expect them to be non-null.